internal/chezmoi/realsystem.go

```diff
@@ -100,10 +100,20 @@ func (s *RealSystem) RunScript(scriptname RelPath, dir AbsPath, data []byte) (er
 	// Run the temporary script file.
 	//nolint:gosec
 	cmd := exec.Command(f.Name())
-	cmd.Dir = string(dir)
 	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
+
+	// The requested working dir may not actually exist yet, if this is a
+	// `before_` script.  In that case, look through the parent hierarchy till
+	// we find one.
+	for workDir := dir; string(dir) != "."; workDir = workDir.Dir() {
+		if _, err = os.Stat(string(workDir)); !os.IsNotExist(err) {
```

Comment:
This silently ignores all non-`os.IsNotExist` errors. Non-`nil` errors that are not `os.IsNotExist` errors should be returned. This also uses `os.Stat` whereas it should use `s.Stat`.

Reply:
Oops, I totally should have picked up on that. Thanks!